src/doc/trpl/dining-philosophers.md

```diff
@@ -2,45 +2,40 @@
-1965, but we’ll use the version from [this paper][paper] by Tony Hoare in 1985.
-
-[paper]: http://www.usingcsp.com/cspbook.pdf
-
-> In ancient times, a wealthy philanthropist endowed a College to accommodate
-> five eminent philosophers. Each philosopher had a room in which she could
-> engage in her professional activity of thinking; there was also a common
-> dining room, furnished with a circular table, surrounded by five chairs, each
-> labelled by the name of the philosopher who was to sit in it. They sat
-> anticlockwise around the table. To the left of each philosopher there was
-> laid a golden fork, and in the centre stood a large bowl of spaghetti, which
-> was constantly replenished. A philosopher was expected to spend most of her
-> time thinking; but when she felt hungry, she went to the dining room, sat down
-> in her own chair, picked up her own fork on her left, and plunged it into the
-> spaghetti. But such is the tangled nature of spaghetti that a second fork is
-> required to carry it to the mouth. The philosopher therefore had also to pick
-> up the fork on her right. When she was finished she would put down both her
-> forks, get up from her chair, and continue thinking. Of course, a fork can be
-> used by only one philosopher at a time. If the other philosopher wants it, she
-> just has to wait until the fork is available again.
+1965, and has been used as an example of the use of concurrency primitives in
+many influential works since then.
+
+Plato, Hypatia, Lao, Simone, and Audre are professors of philosophy at a
+university. As philosophers, they spend most of their time in their offices
+thinking and writing; but occasionally, they need to come out for food. There
+is a round table in the common room between their offices, and they each have
+a designated seat at it; the university ensures that there is always a fresh
+bowl of rice at each seat, so that they may come out and dine whenever they
+please. However, due to budgetary constraints, the university has only been
+able to afford five chopsticks for the five professors; in order to ensure
+that everyone can eat, the chopsticks are arranged between the places at the
+table, so each professor can pick up the two adjacent chopstics and begin
+eating. When each is done eating, she puts down her chopsticks in the
+original locations, and returns to her studies.
 
 This classic problem shows off a few different elements of concurrency. The
 reason is that it's actually slightly tricky to implement: a simple
 implementation can deadlock. For example, let's consider a simple algorithm
 that would solve this problem:
 
-1. A philosopher picks up the fork on their left.
-2. They then pick up the fork on their right.
-3. They eat.
-4. They return the forks.
+1. A philosopher picks up the chopstick on her left.
+2. She then picks up the chopstick on her right.
+3. She eats.
+4. She returns the chopsticks.
```

Comment:
> Never in English has a female pronoun been acceptable for a heterogeneous collection with items of both genders, like male pronouns have been (over the past couple of centuries in particular this has been becoming less acceptable).

"Never" is a very strong word, and thus very easy to refute. [Here's one example](http://journals.lww.com/ajnonline/Fulltext/2001/03000/PRONOUN_EQUALITY.7.aspx) in which the American Journal of Nursing uses "she" to refer to both nurses and physicians, both groups of mixed gender (here's a [specific example](http://journals.lww.com/ajnonline/Citation/1979/12000/BURNOUT__The_Elaboration_of_a_Concept_.20.aspx)).

It is very common to use "she" for unspecified people of a profession or engaged in an activity that is coded to be "women's work," while "he" for people engaged in work that is coded to be "men's work." There is nothing linguistically invalid about using "she" for an unspecified member of a mixed group; it is merely surprising due to that cultural bias about what constitutes men's vs. women's work.

By deliberately choosing to use feminine pronouns to refer to a mixed group (of predominantly women) of a profession that has been traditionally coded as "men's work", it can help expose those cultural biases, make you stop and think rather than silently reinforcing them.